File: VSCode_Java/leetcode/editor/en/494.target-sum.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    int ans = 0;

    private void backtrack(int[] nums, int i, int val, int target) {
        if (i == nums.length) {
            if (val == target) {
                ans ++;
            }
            return;
        }

        backtrack(nums, i+1, val+nums[i], target);
        backtrack(nums, i+1, val-nums[i], target);
        return;

    }

    public int findTargetSumWays(int[] nums, int target) {
        int n = nums.length;
        backtrack(nums, 0, 0, target);
        return ans;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: 22FALL/494.target-sum.py
#
# @lc app=leetcode id=494 lang=python3
#
# [494] Target Sum
#

# @lc code=start
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        self.ans = 0

        def backtrack(i, val):
            if i== len(nums):
                if val == target:
                    self.ans += 1
                return
            backtrack(i+1, val + nums[i])
            backtrack(i+1, val - nums[i])
            return
        
        backtrack(0, 0)
        return self.ans

# @lc code=end

